include/zephyr/bindesc.h

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright (c) 2023 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZEPHYR_BINDESC_H_
+#define ZEPHYR_INCLUDE_ZEPHYR_BINDESC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/*
+ * Corresponds to the definitions in scripts/west_commands/bindesc.py.
+ * Do not change without syncing the definitions in both files!
```

Comment:
It'd be great if the Python script parsed this header instead.

Reply:
My concern here was about how the script would find that file. Looking for `ZEPHYR_BASE` might not work because the user may have another instance of Zephyr of an older version, and looking for it relative to the script may also not work if the script is running from a copied version (e.g. west is running from a venv)

Reply:
I don't think that running from a venv changes the location of the original script, only those of the pip modules that you are installing in the venv. But let's leave this as an improvement.